src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```
@@ -0,0 +1,377 @@
+          new TableRow().set(REFERENCE_NAME,
+                             c.element().getAlignment().getPosition().getReferenceName())
+                        .set(POSITION, c.element().getAlignment().getPosition().getPosition())
+                        .set(REVERSE_STRAND,
+                             c.element().getAlignment().getPosition().getReverseStrand());
+      List<TableRow> alignmentCigars = new ArrayList<>();
+      for (CigarUnit cigar : c.element().getAlignment().getCigarList()) {
+        TableRow alignmentCigar =
+            new TableRow().set(OPERATION, cigar.getOperation().getNumber())
+                          .set(OPERATION_LENGTH, cigar.getOperationLength())
+                          .set(REFERENCE_SEQUENCE, cigar.getReferenceSequence());
+        alignmentCigars.add(alignmentCigar);
+      }
+      TableRow alignmentRecord = new TableRow().set(POSITION, alignmentPosition)
+          .set(MAPPING_QUALITY, c.element().getAlignment().getMappingQuality())
+          .set(CIGAR, alignmentCigars);
+      row.set(ALIGNMENT, alignmentRecord);
+
+      c.output(row);
+    }
+  }
+
+  public static void main(String[] args) throws GeneralSecurityException, IOException,
+         URISyntaxException {
+    // Register the options so that they show up via --help
+    PipelineOptionsFactory.register(Options.class);
+    pipelineOptions = PipelineOptionsFactory.fromArgs(args)
+        .withValidation().as(Options.class);
+    // Option validation is not yet automatic, we make an explicit call here.
+    Options.Methods.validateOptions(pipelineOptions);
+    // Needed for BigQuery.
+    pipelineOptions.setTempLocation(pipelineOptions.getTempPath());
+
+    auth = GenomicsOptions.Methods.getGenomicsAuth(pipelineOptions);
+    p = Pipeline.create(pipelineOptions);
+
+    // Ensure data is accessible
+    String BAMFilePath = pipelineOptions.getBAMFilePath();
+    if (!Strings.isNullOrEmpty(BAMFilePath)) {
+      if (GCSURLExists(BAMFilePath)) {
```

This function swallows the error so it's impossible to say whether the file doesn't exist, a transient error occurred, or permissions were denied, for example.

---

I added print-the-exception inside GCSURLExists(). Let me know if you prefer something else.